### PR TITLE
Add object-state plugin

### PR DESCRIPTION
**Pull request by chrisdevereux**

### CODE REVIEW

**Review by wheelr — APPROVED** [review]
Looks fine but I'm not clear about the 'why'.

File: src/lib/plugins/ObjectStatePlugin/ObjectStateActions.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      return id in this.objects.value
+    }
+
+    query(predicate: (x: T) => boolean): T[] {
+      return filter(this.objects.value, (x) => Boolean(x && predicate(x))) as T[]
+    }
+
+    create(id: string, data: T): string
+    create(data: T): string
+    create(idOrData: any, data?: any) {
```
Comment (robotdevil) [inline src/lib/plugins/ObjectStatePlugin/ObjectStateActions.ts:46]: Anything that can be done to break this up into sub functions that explain what it's doing a bit more clearly?

**Comment by robotdevil** [comment]
Outside of the one comment about the create method, I'm fine with this - like @wheelr I'm not clear on what it's for though?

**Comment by chrisdevereux** [comment]
Closing this after discussion that this plugin isn't that useful and adds unneeded complexity

**Pull request closed without merging**